src/libs/BouncerProvider.js

```diff
@@ -196,6 +199,11 @@ export default class BouncerProvider {
         }
 
         this.state.networks.forEach((network) => {
+            // don't save the controller network or add it to the bnc
+            if (controller === network) {
```

Comment:
Alright so the issue with this method is that the controller network can **either** be a special separate network (e.g. when you first join, which we don't want to save) or it can be a regular bouncer network that's connected to an upstream network (which we do want to save).

What this should do is check the `BOUNCER` isupport token for a value (but I need to make kiwibnc send that earlier so this check is more accurate and happens at the right time).

Reply:
There's a current todo item in the bnc for including custom isupport tokens in a similar fashion to caps, but I'm not sure how that would effect this method in BouncerProvider?

Also, what's this special network thing?

Reply:
tl;dr If the controller network is not connected to an upstream (e.g. the automatic controller network that's created when we first join KiwiIRC), then that connection should not be saved to the bouncer. If it does get saved to the bouncer, the bnc has a new network added to it whose details are "Connect to <bouncer_address> on <bouncer_port>" and that's it, which doesn't work and is why we try to avoid this case.

If the controller network is connected to an upstream, then we can save it here without any issues. This piece of code is fixed by the latest commit, and depends on the improved `005` sending in this bouncer PR: https://github.com/kiwiirc/kiwibnc/pull/11